torch/cuda/__init__.py

```diff
@@ -922,6 +997,66 @@ def _get_pynvml_handler(device: Optional[Union[Device, int]] = None):
     return handle
 
 
+def _get_amdsmi_handler(device: Optional[Union[Device, int]] = None):
+    if not _HAS_PYNVML:
+        raise ModuleNotFoundError(
+            "amdsmi does not seem to be installed or it can't be imported."
+        ) from _PYNVML_ERR
+    try:
+        amdsmi.amdsmi_init()
+    except amdsmi.AmdSmiException as e:
+        raise RuntimeError("amdsmi driver can't be loaded, is ROCm installed?") from e
```

Comment:
Maybe fall back or add a suggestion on RoCM version on amdsmi supports (like successor of rocm smi)?

Reply:
Adopted suggestion on ROCm version